### PR TITLE
chore(22552): refactor modal selector in e2e tests

### LINKED ISSUE #22552: Refactor to use test ID or text selector for modal selection in e2e https://github.com/MetaMask/metamask-extension/pull/22548#discussion_r1453897798

### PR DESCRIPTION
**Pull request by DDDDDanica**
## **Description**
Following the suggestion divided from [this thread](https://github.com/MetaMask/metamask-extension/pull/22548#discussion_r1453897798) to refactor all selectors for modal in e2e tests. They are now referred by `data-testid` in selector. 
More specifically speaking, when the modal has been adapted to new component, a `data-testid` will be appended directly to `<Modal >` component and selected; when the component is using deprecated modal component, this `data-testid` will be appended to as a property passing to `FadeModal` component 
  

## **Related issues**

Fixes: #22552

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by HowardBraham** [comment]
You have a few instances of `('[data-testid="account-details-modal"')` with unbalanced square brackets

**Comment by DDDDDanica** [comment]
@HowardBraham Thanks a lot, that's a good catch. 👍🏻

File: test/e2e/webdriver/driver.js [hunk 1]
```diff
@@ -79,17 +79,22 @@ function wrapElementWithAPI(element, driver) {
...
+        if (e.message.includes('<div class="mm-box loading-overlay">')) {
+          // Wait for the loading overlay to disappear and try again
+          await driver.wait(
+            until.elementIsNotPresent(By.css('.loading-overlay')),
+          );
+        }
+        if (e.message.includes('<div class="modal__backdrop">')) {
+          // Wait for the modal to disappear and try again
+          await driver.wait(
+            until.elementIsNotPresent(By.css('.modal__backdrop')),
```
Comment (DDDDDanica) [inline test/e2e/webdriver/driver.js:92]: Similar methodology as https://github.com/MetaMask/metamask-extension/pull/22672, we would wait for modal to be completed removed and then perform clicking actions.

PS: Once we completely migrated modal component, we can remove this await.

**Review by chloeYue — APPROVED** [review]
LGTM!

**Review by HowardBraham — APPROVED** [review]

**Pull request merged**